lib/nvmf/ctrlr.c

```diff
@@ -209,6 +209,8 @@ nvmf_ctrlr_keep_alive_poll(void *ctx)
 					      nvmf_ctrlr_disconnect_qpairs_on_pg,
 					      ctrlr,
 					      nvmf_ctrlr_disconnect_qpairs_done);
+			if(ctrlr->subsys->nvmf_ss_event_cb)
```

Comment:
I'd suggest moving event propagation into a separate function, like this:
```__notify_subsystem_event(ctrlr, event)```

and move all the code to it:
```
if(subsystem->nvmf_ss_event_cb)
	subsystem->nvmf_ss_event_cb
```

Then just call it for required events: __notify_subsystem_event(ctrlr, SPDK_NVMF_SS_INIATOR_TIMEOUT)

Reply:
We can do that but the event propagation is one line, other than checking for the non-NULL callback function. Or, am I missing something?

Reply:
Having a generic function for event propagation allows us, for instance, add logging (in. one place), add some debug info just once instead of copying that to every place where we perform notification. Makes life easier from maintainability perspective,  IMHO.

Reply:
Sure, added a wrapper around the callback! BTW, ctrlr is an event-specific parameter, hence NULL check can't be inside the wrapper

Reply:
created an inline function